### ISSUE TITLE
TextInput not respecting width

### ISSUE THREAD

**Issue by auroraTDunn** [bokeh/bokeh#6427]
Looks like TextInput is still not respecting 'width' in v0.12.5.

```
from bokeh.plotting import figure, show, curdoc, output_notebook
from bokeh.layouts import widgetbox, layout
from bokeh.models.widgets import TextInput

wd = 20
txtIn_start = TextInput(value=str(1), title="Start:", width=wd)
txtIn_end   = TextInput(value=str(9), title="End:", width=wd)
txtIn_step  = TextInput(value=str(3), title="Step:", width=wd)

txtIn_rng = widgetbox(children=[txtIn_start,txtIn_end,txtIn_step], width=wd)

#show(txtIn_rng_row)
curdoc().add_root(txtIn_rng)
```

![textinput_width](https://user-images.githubusercontent.com/2123822/26989891-0200f2e0-4d12-11e7-8431-020ada9b3fea.png)

**Comment by auroraTDunn** [comment]
Looks like TextInput is still not respecting 'width' in v0.12.5.


from bokeh.plotting import figure, show, curdoc, output_notebook
from bokeh.layouts import widgetbox, layout
from bokeh.models.widgets import TextInput

wd = 20
txtIn_start = TextInput(value=str(1), title="Start:", width=wd)
txtIn_end   = TextInput(value=str(9), title="End:", width=wd)
txtIn_step  = TextInput(value=str(3), title="Step:", width=wd)

txtIn_rng = widgetbox(children=[txtIn_start,txtIn_end,txtIn_step], width=wd)

#show(txtIn_rng_row)
curdoc().add_root(txtIn_rng)


![textinput_width](https://user-images.githubusercontent.com/2123822/26989891-0200f2e0-4d12-11e7-8431-020ada9b3fea.png)

**Comment by rmchurch** [comment]
Temporary workaround until fixed in 0.12.7 or .8:
Place this in a custom CSS file:

//this overwrites TextInput class, so width works correctly
.bk-widget-form-input {
    width: 600px; //!important
}

**Comment by ManchesterWuer** [comment]
@rmchurch Would you please explain in a more detail way to how you did this? I am now using Bokeh 0.12.10 and still can't change the size of TextInput. You mentioned a styles.css sheet, do I need to write these codes directly into the same .py file of the TextInput, or I need to write these codes in a separate .css file?

**Comment by rmchurch** [comment]
@ManchesterWuer I run it separate (maybe you could figure out how to include in the main py file). You have to run the bokeh server [https://bokeh.pydata.org/en/latest/docs/user_guide/server.html](url)
So a simple example would be have a directory with the following:
myapp
|
+--main.py
|--templates
     +--styles.css

and you would run it using "bokeh serve myapp"

**Comment by ManchesterWuer** [comment]
@rmchurch I just repeated what you did to control the size of TextInput, using Bokeh 0.12.13. Unfortunately, I still got the unchanged size as the original TextInput. Did I miss something? Are there any other codes I should put into styles.css other than those 3 lines of codes you offered?

**Comment by rmchurch** [comment]
No, that should be it. Perhaps post simplified code that reproduces, with how exactly you are running it, or post on the gitter/SO

**Comment by timothydmorton** [comment]
I am having this same problem.  Here is my minimal example:
```python
from bokeh.io import curdoc
from bokeh.models import TextInput

def modify_doc(doc):
    text_box = TextInput(value='this text is longer than 300px', title='input',
                         css_classes=['customTextInput'])

    doc.add_root(text_box)
    return doc


doc = modify_doc(curdoc()) 
```
And here is my directory:
```
12:46 $ tree textinput-test
textinput-test
├── main.py
└── templates
    └── styles.css
```
and `styles.css` file:
```
12:46 $ cat textinput-test/templates/styles.css
.customTextInput .bk-widget-form-input {
    width: 600px;
}
```
I'm running Bokeh 0.12.13 on macOS, conda python 3.6.1

**Comment by ManchesterWuer** [comment]
@rmchurch Are you using Mac os? I guess maybe Windows is the reason.

**Comment by timothydmorton** [comment]
Posted to SO: https://stackoverflow.com/questions/48172459/bokeh-textinput-width-keyword-not-working

**Comment by rmchurch** [comment]
FWIW, I just ran this on MacOS also, and got your result (i.e. width not updated). The fix I posted is working on Linux, not sure how to adapt for Mac.

**Comment by rmchurch** [comment]
Ah sorry, I lead you astray. I also have an index.html, to use the styles.css. Add this to your templates directory:
```
<!DOCTYPE html>
<html lang="en">
    <head>
        <script src="https://code.jquery.com/jquery-3.1.0.min.js"></script>
        <style>
            @import url(https://fonts.googleapis.com/css?family=Noto+Sans);
            body {
              font-family: 'Noto Sans', sans-serif;
              -webkit-font-smoothing: antialiased;
              text-rendering: optimizeLegibility;
              color: #fff;
              background: #2F2F2F;
             }
        </style>
        <meta charset="utf-8">
        <title>Bokeh Crossfilter Example</title>
        {{ bokeh_css }}
        {{ bokeh_js }}
        <style>
            {% include 'styles.css' %}
        </style>
    </head>
    <body>
        {{ plot_div|indent(8) }}
        {{ plot_script|indent(8) }}
    </body>
</html>
```

**Comment by timothydmorton** [comment]
Wonderful, thank you!

**Comment by bryevdv** [comment]
This appears to have been fixed as of #8085 

Original code with `wd=20` and then `wd=200`

<img width="328" alt="screen shot 2019-02-18 at 16 49 51" src="https://user-images.githubusercontent.com/1078448/52982881-60e4cc00-339d-11e9-8cef-e76c8960c2c1.png">

<img width="245" alt="screen shot 2019-02-18 at 16 49 41" src="https://user-images.githubusercontent.com/1078448/52982884-63472600-339d-11e9-9c44-aa12308e299d.png">
